tests/helpers/perf_run_helper.rb

```diff
@@ -944,75 +945,253 @@ def get_process_hash(perf_result_processes)
+  def find_atop_log_from_dir(dir, runtype)
+    find_file(dir, "atop_log_#{runtype}*.csv")
+  end
+
+  # @param  [String] csv_string  CSV string from atop file created by BeakerBenchmark PerformanceResult.log_csv.
+  #
+  # @return [Hash{global=>Hash,processes=>Hash}]  Hash of atop hashes
+  def result_hashes_from_atop_csv(csv_string)
+    tmp = csv_string.split("\n\n")
+    # Tease out global results
+    gbl_tbl = CSV.parse(tmp[0], headers: true)
+    gbl_h = gbl_tbl.first.to_h
+    # Tease out process results
+    process_tbl = CSV.parse(tmp[1], headers: true)
+    process_h = {}
+    process_tbl.each do |r|
+      h = r.to_h
+      d = { cmd: h["command"],
+            avg_cpu: h["Avg CPU"].to_i,
+            avg_mem: h["Avg MEM"].to_i,
+            avg_disk_read: h["Avg DSK read"].to_i,
+            avg_disk_write: h["Avg DSK Write"].to_i }
+      process_h[h["Process pid"]] = d
+    end
+    { global: gbl_h, processes: process_h }
+  end
+
+  # @param  [String] dir  Path where results are stored
+  # @param  [String] runtype  Test type used to generate results
+  #
+  # @return [PerformanceResult]
+  def atop_results_from_dir(dir, runtype)
+    # This is a dirty, dirty reconstitution of the data from a CSV file
+    # created by [beaker-benchmark](https://github.com/puppetlabs/beaker-benchmark/blob/master/lib/beaker-benchmark/helpers.rb#L219-L231).
+    file = find_atop_log_from_dir(dir, runtype)
+    csv_string = read_file(file)
+    atop_hashes = result_hashes_from_atop_csv(csv_string)
+
+    # Create PerformanceResult object
+    perf_results_args = {
+      cpu: [],
+      mem: [],
+      disk_read: [],
+      disk_write: [],
+      action: atop_hashes[:global]["Action"],
+      duration: atop_hashes[:global]["Duration"].to_f.round(2),
+      processes: {},
+      logger: nil,
+      hostname: Pathname(file).parent.split[1].to_s
+    }
+    atop_res = Beaker::DSL::BeakerBenchmark::Helpers::PerformanceResult.new(perf_results_args)
+    atop_res.avg_cpu = atop_hashes[:global]["Avg CPU"].to_i
+    atop_res.avg_mem = atop_hashes[:global]["Avg MEM"].to_i
+    atop_res.avg_disk_read = atop_hashes[:global]["Avg DSK read"].to_i
+    atop_res.avg_disk_write = atop_hashes[:global]["Avg DSK Write"].to_i
+    atop_res.processes = atop_hashes[:processes]
+    atop_res
+  end
+
+  # @param  [String] dir  Path to look for file in
+  # @param  [String] pat  Regular expression pattern to find file with
+  #
+  # @return [String]  Path to file
+  def find_file(dir, pat)
+    file = `find #{dir} -name "#{pat}" -print`.chomp
+    raise System.StandardError "The file does not exist" unless File.exist?(file)
+
+    file
+  end
+
+  # @param  [String] dir  Path where gating assertions are stored
+  #
+  # @return [String]  Path to assertions file
+  def find_gatling_assertions_from_dir(dir)
+    find_file(dir, "assertions.json")
+  end
+
+  # @param  [String] file  File to read from
+  #
+  # @return [String]  Contents of file
+  def read_file(file)
```

Comment:
why?

Reply:
I used this as a simple wrapper in an effort to make mocking it in the spec tests easier.  It may not be necessary.  I will investigate.

Reply:
like this one...
allow(subject).to receive(:read_file).with("atop_file").and_return(atop_file_content)
you can do this with
allow(File).to receive(:read).with("atop_file").and_return(atop_file_content)
Then you won't need the wrapper.

Reply:
Wrapper deleted.  Thanks for the example.